contrib_bots/lib/github_issues.py

```diff
@@ -0,0 +1,149 @@
+
+
+
+class IssueHandler(object):
+    '''
+    This plugin facilitates sending issues to github, when
+    an item is prefixed with '@issue' or '@bug'
+
+    It will also write items to the issues stream, as well 
+    as reporting it to github
+    '''
+
+    def usage(self):
+        return '''
+            This plugin will allow users to flag messages
+            as being issues with Zulip by using te prefix '@issue'
+            
+            Before running this, make sure to create a stream
+            called "issues" that your API user can send to.
+            
+            Also, make sure that the credentials of the github bot have
+            been typed in correctly, that there is a personal access token
+            with access to public repositories ONLY, 
+            and that the repository name is entered correctly.
+            '''
+
+      
+    
+    def triage_message(self, message):
+        # return True if we want to (possibly) respond to this message
+
+        original_content = message['content']
+
+        # This next line of code is defensive, as we
+        # never want to get into an infinite loop of posting follow
+        # ups for own follow ups!
+        if message['display_recipient'] == 'issue':
+            return False
+        is_issue = (original_content.startswith('@issue'))
+
+        return is_issue
+
+    def handle_message(self, message, client, state_handler):
+        original_content = message['content']
+        original_sender = message['sender_email']
+        
+        print('sending issue')
+        #gets token from google drive
+        data = urllib2.urlopen("https://drive.google.com/uc?export=download&id=0Bwo9Sux-T0A9QVFKV2FGb1RxbnM") #Downloads the file containing the token
+        for line in data: # files are iterable
+            print line
+            PASSWORD = line
+        # Authentication for user filing issue DOES NOT WORK IF
+        # THE USER HAS READ/WRITE ACCESS TO THE REPOSITORY
+        USERNAME = 'zulip-issue-bot'
+       
+        #824d6ae0b791e0b6de9a3169d2f6ecd193f66ebe
+
+
+        # The repository to add this issue to
+        REPO_OWNER = 'fredfishgames'
+        REPO_NAME = 'zulip'
+        
+        
+        new_content = original_content.replace('@issue',
+                                               'by %s:' % (original_sender,))
+        
+        
+        #gets the repo url
+        url = 'https://api.github.com/repos/%s/%s/issues' % (REPO_OWNER, REPO_NAME)
+        
+        #signs into github using the provided username and password
+        session = requests.session()
+        session.auth=(USERNAME, PASSWORD)
+        print('Logged in to github as %s' % (USERNAME))
+        #is_tag = False
+        #labels = ["from bot"]
+        
+        #current_tag = ""
+        
+        #Gets rid of the @issue in the issue title
+        issueTitle = message['content'].replace('@issue ', '')
+        issueTitle = message['content'].replace('@issue', '')
```

Comment:
Change `issueTitle` to `issue_title` and use strip() `message['content'].replace('@issue', '').strip()`. I think you would probably want to limit length of title as well.